app/service_invite/rest.py

```diff
@@ -121,3 +124,95 @@ def validate_service_invitation_token(token):
+        message = f"You are already a member of {service.name}"
+        raise BadRequestError(message=message)
+
+    # Send the user's service invite request to the service managers listed
+    send_service_invite_request(
+        user_requesting_invite, recipients_of_invite_request, service, reason_for_request, accept_invite_request_url
+    )
+
+    # Send a receipt email to the user that requested the invite
+    send_receipt_after_sending_request_invite_letter(user_requesting_invite)
+
+    return {}, 204
+
+
+def send_service_invite_request(
+    user_requesting_invite, recipients_of_invite_request, service, reason_for_request, accept_invite_request_url
+):
+    template_id = current_app.config["REQUEST_INVITE_TO_SERVICE_TEMPLATE_ID"]
+    template = dao_get_template_by_id(template_id)
+    notify_service = Service.query.get(current_app.config["NOTIFY_SERVICE_ID"])
+    number_of_notifications_generated = 0
+    for recipient in recipients_of_invite_request:
+        if service in recipient.services:
+            saved_notification = persist_notification(
+                template_id=template.id,
+                template_version=template.version,
+                # TODO change recipient to actual email address of service managers when the testing phase completes
+                recipient="notify-join-service-request@digital.cabinet-office.gov.uk",
+                service=notify_service,
+                personalisation={
+                    "name": user_requesting_invite.name,
+                    "requester_name": user_requesting_invite.name,
+                    "requester_email": user_requesting_invite.email_address,
+                    "service_name": service.name,
+                    "reason_given": "yes" if reason_for_request else "no",
+                    "reason": reason_for_request,
+                    "url": accept_invite_request_url,
+                },
+                notification_type=template.template_type,
+                api_key_id=None,
+                key_type=KEY_TYPE_NORMAL,
+                reply_to_text=notify_service.get_default_reply_to_email_address(),
+            )
+            send_notification_to_queue(saved_notification, queue=QueueNames.NOTIFY)
+            number_of_notifications_generated += 1
+
+        else:
+            # In a scenario were multiple service managers are listed, and the list contains an
+            # invalid service manager, we would rather log the errors and not raise an exception so
+            # that notifications can still be sent to the valid service managers
+            message = f"Can’t create notification - {recipient.name} is not part of the {service.name}"
+            current_app.logger.error(message)
+
+    if number_of_notifications_generated == 0:
+        # If no notification is sent we want to raise an exception
+        message = f"Can’t create notification as the service manager listed is not part of the {service.name}"
```

Comment:
Again, I think let's come up with an error code for this and use it instead. Maybe something like `no-valid-service-managers`?